### ISSUE TITLE
Optionally require email addresses to be verified

### ISSUE THREAD

**Issue by alfongj** [firebase/FirebaseUI-Android#312]
Provide optional setting to require email addresses to be verified in order for a user to sign in with email and password.

**Comment by SUPERCILEX** [comment]
This can be done with [Play Services 9.6](https://firebase.google.com/support/release-notes/android#9.6) using `FirebaseUser#sendEmailVerification()` and `FirebaseUser#isEmailVerified()`.

**Comment by samtstern** [comment]
@SUPERCILEX good point.  I think @alfongj was hoping we could provide this as a `SignInIntentBuilder` option to fail sign in for unverified email addresses and have some UI to send the email automatically.

**Comment by SUPERCILEX** [comment]
Yeah, I just wanted to point out an easy way to handle the verification for anyone submitting a PR. The UI part seems like it will be difficult because we would need to check periodically with isEmailVerified and not let let the user in if it isn't. Anyway, food for thought.

**Comment by Eragon29** [comment]
Hm, was this suggestion about the automated verification email, or more like if you have no invite, you can’t create an account?

**Comment by samtstern** [comment]
@Eragon29 I think the suggestion to to help with the entire flow:

  * If a user tries to sign in but has an unverified email, send them the verification and show some "please check your email screen"
  * When the user has verified and returns to the app, we should detect verification and allow them to proceed.

**Comment by Eragon29** [comment]
@samtstern
Too bad
Would be nice to have a way to manually verify emails. But it’s probably not that important and there are ways to work around this “issue”

**Comment by baole** [comment]
It would be nice to request email verification in the AuthUI builder, such as

new AuthUI.IdpConfig.EmailBuilder()
                                            .setRequireName(false)
                                            **.setRequireEmailVerification(true)** // not in the builder atm
                                            .build()

And this library will handle the rest.

**Comment by AndrazP** [comment]
What is the status of this feature request after almost 4 years?
I just started integrating Firebase Auth and was really surprised after finding out verification emails are not sent out automatically.
It's a basic functionality that Firebase Auth should offer effortlessly.

**Comment by samtstern** [comment]
@AndrazP no update, I do think this would be useful it's just that FirebaseUI has been focused on stability and adding support for more identity providers for the last few big releases.

**Comment by demolaf** [comment]
Hello,

Thank you for taking the time to submit this feature request.

We are happy to let you know that this has been addressed in **FirebaseUI for Android v10.0.0**, which is now available as a beta release. You can find the release notes [here](https://github.com/firebase/FirebaseUI-Android/releases/tag/10.0.0-beta02) and follow the [upgrade guide](https://github.com/firebase/FirebaseUI-Android/blob/master/docs/upgrade-to-10.0.md) to get started.

We will close this feature request now. If this is not fully addressed in the new version, please feel free to open a new issue.

Thank you again for your contribution.
